File: studies/movies/ggplot2/scatter_connected.R
library(tidyverse)
library(lubridate)

df <- read_csv("data/movies.csv") |>
  group_by(year(`Release Date`)) |>
  summarize_at(vars(`Production Budget`, `Worldwide Gross`), ~ sum(.x, na.rm = TRUE))

ggplot(df) +
  aes(
    x = `Production Budget`, y = `Worldwide Gross`
  ) +
  geom_point() +
  geom_path()

File: galleries/movies/pandas/scatter_connected.py
import pandas as pd

df = pd.read_csv("data/movies.csv", parse_dates=["Release Date"])
df["Release Date"] = df["Release Date"].dt.year
ax = (
    df.groupby("Release Date")[["Worldwide Gross", "Production Budget"]]
    .sum()
    .reset_index()
    .plot.line(x="Production Budget", y="Worldwide Gross", style="o-")
)
